src/Test/ProxyTestCase.php

```diff
@@ -64,32 +68,34 @@ public static function isCacheMiss()
     /**
      * Get HTTP response from your application
      *
-     * @param string $url
-     * @param array  $headers
-     * @param array  $options
+     * @param string $uri     HTTP URI
+     * @param array  $headers HTTP headers
+     * @param string $method  HTTP method
      *
-     * @return Response
+     * @return ResponseInterface
      */
-    public function getResponse($url, array $headers = array(), $options = array())
+    public function getResponse($uri, array $headers = [], $method = 'GET')
     {
-        return $this->getHttpClient()->get($url, $headers, $options)->send();
+        // Close connections to make sure invalidation (PURGE/BAN) requests will
+        // not interfere with content (GET) requests.
+        $headers['Connection'] = 'Close';
+        $request = $this->createRequest($method, $uri, $headers);
```

Comment:
This is the test HTTP adapter (`getHttpAdapter()`), which fakes requests to the application, not the caching proxy (`getProxyClient()`) that sends invalidation requests. So I don’t think it should go through the proxy client’s RequestQueue.